aten/src/ATen/native/TensorProperties.cpp

```diff
@@ -53,10 +53,19 @@ Tensor & detach_(Tensor & self) {
 }
 
 Tensor contiguous(const Tensor & self) {
-  if (self.is_contiguous()) {
+  return contiguous(self, MemoryFormat::Contiguous);
+}
+
+Tensor contiguous(const Tensor & self, MemoryFormat memory_format) {
+  if (self.is_contiguous(memory_format)) {
     return self;
   }
-  return self.clone();
+  auto result = at::empty_like(self);
+  if (memory_format == MemoryFormat::ChannelsLast) {
+    AT_CHECK(self.dim() == 4, " required rank 4 tensor to use channels_last format");
+    result.maybe_as_channels_last();
```

Comment:
I guess it's OK to discard the returned bool here, because you checked that dim = 4... but then you should have an ASSERT to that effect, no?